pkg/apis/internal.admin.acorn.io/v1/resources.go

```diff
@@ -75,31 +107,21 @@ func (current *Resources) Remove(incoming Resources, all bool) {
 func (current *Resources) Fits(incoming Resources) error {
 	exceededResources := []string{}
 
-	// Define function for checking int resources to keep code DRY
-	checkResource := func(resource string, currentVal, incomingVal int) {
-		if currentVal <= incomingVal {
-			exceededResources = append(exceededResources, resource)
+	for count, value := range incoming.Counts {
+		if current.Counts[count] < value {
+			exceededResources = append(exceededResources, string(count))
 		}
 	}
-
-	// Define function for checking quantity resources to keep code DRY
-	checkQuantityResource := func(resource string, currentVal, incomingVal resource.Quantity) {
-		if currentVal.Cmp(incomingVal) <= 0 {
-			exceededResources = append(exceededResources, resource)
+	for persistentCount, value := range incoming.PersistentCounts {
+		if current.PersistentCounts[persistentCount] < value {
```

Comment:
nit: I noticed that the previous code was using `<=` here, should that be the case for the new code as well?

Reply:
Yeah so I thought about this when doing this refactor. I think it should actually be `<` because if it is `<=` then this case will fail if the quantity we're comparing to equal. This is not desirable since that is actually a success case. This lead me to add a test suite. Let me know what you think.